NN_training_example.py

```diff
@@ -26,28 +26,35 @@
           np.arange(-params.matrix_size / 2, params.matrix_size / 2) * params.pixel] for y in
          np.arange(-params.matrix_size / 2, params.matrix_size / 2) * params.pixel])
 
-    #Build NNTrainer
+    # Build NNTrainer
     NN = NNTrainer(params)
 
     # Run NN optimization
+    # Please try running different numbers of epochs (last parameter)
+    # Check the difference in the output for different amounts of training
     trained_model = NN.optimize(LightField(amp, phase), LightField(target, phase), 100)
 
     # Plot loss vs epochs
     NN.plot_loss()
 
-    #Plot the result - optimized phase map
-    plotter = GeneratePropagationPlot(LightField(amp, np.array(trained_model.layers[3].get_weights()[0])), output_type=GeneratePropagationPlot.PLOT_PHASE)
+    # Plot the result - optimized phase map
+    plotter = GeneratePropagationPlot(LightField(amp, np.array(trained_model.layers[3].get_weights()[0])),
+                                      output_type=GeneratePropagationPlot.PLOT_PHASE)
     plotter.save_output_as_figure("outs/NNstructure.png")
 
+    # Plot the target amplitude
     plotter = GeneratePropagationPlot(LightField(target, phase), output_type=GeneratePropagationPlot.PLOT_ABS)
     plotter.save_output_as_figure("outs/NNtarget.png")
 
+    # Plot the input amplitude
     plotter = GeneratePropagationPlot(LightField(amp, phase), output_type=GeneratePropagationPlot.PLOT_ABS)
     plotter.save_output_as_figure("outs/NNinput.png")
 
+    # Plot the result - output amplitude
     field = np.array([amp, phase])
     field = field.reshape((1, 2, params.matrix_size, params.matrix_size,), order='F')
     result = trained_model(field).numpy()
     result = result[0, 0, :, :] * np.exp(1j * result[0, 1, :, :])
```

Comment:
Tbh I have no idea what's going on here on the first glance. At least I'd wrap it within a function with more descriptive name. Preferably in the long run I'd analyze why is this even needed and how we can get rid of it - I mean not the functionality itself, but how its exposed to the end user.

Reply:
I agree. In my opinion, it should be more general method for importing trained models and evaluating them. Created a separate issue. For now additional comments will do.